unit2-apis-and-advanced-javascript/README.md

```diff
@@ -18,68 +18,68 @@
+
 
 ### Week 2
 
-- Scope: .5 day
-  - Hoisting 
-  - Investigating the scope chain
+- Scope & Closures: 2 day
+  - Hoisting
+  - Investigating the scope chain, the true power of the `var` keyword
+  - What are closures and how/when/why to create them
   - **Learning outcome**: Be able to answer common "what does this code output" tricky JS questions involving hoisting, scope. Examples can be seen here: https://www.sitepoint.com/5-typical-javascript-interview-exercises/ 
-- Closures: .5 day
-  - What are closures?
-  - How can they be used and when are they useful?
   - **Learning outcome**: Solve tricky problems involving closures, [Examples here](https://github.com/C4Q/ac-curriculum-web/tree/master/unit2-apis-and-advanced-javascript/closure-challenges.md)
 - Prototypes, the prototype chain, and inheritance: 2 days
   - constructor vs `Object.create()`
   - creating base classes and extending them via inheritance
-  - **Learning outcome**: Be able to discuss OOP interview questions (how to design games like chess, model animal species, etc.) and understand how to discuss the talking points for non-JS OOP 
-  - **Learning outcome**: Given a class called Square and two instances of this class, s and t, how would you define `calcArea()` and `calcPermiter()` methods such that _they can be called_ on the two instantiated instances, s and t?
-  - **Learning outomce**: How would you reimplement jQuery from scratch using objects? (ie: `$('..')` gets us an object with methods such as `.css()`. How would you define this c;ass such that it doesn't require a `new` keyword *but still* consoles 'jQuery' when `console.log( $('...') instanceof jQuery )`
-  - **Learning outcome**: How would you extend you jQuery class such that it supports the creation of "plugins" (ie: a new method called `.fooPlugin()` that works seamlessly on all current and future instances of your jQuery obejct) _without touching your_ original jQuery class implementation source code
- 
-- APIs via AJAX and JSON: 2 days
+  - OOP, Prototypical OOP VS traditional class-based OOP, non-JS OOP (Interfaces, Abstract Classes)
+  - **Learning outcome**: Be able to discuss OOP interview questions (how to design games like chess, model animal species, etc.) and understand how to discuss the talking points for non-JS OOP
+  - **Learning outcome**: Design and implement classes in JS, create instances manually and dynamically, and utilize protypical inheritance to augment existing classes on the fly. [Examples here](https://github.com/C4Q/ac-curriculum-web/tree/master/unit2-apis-and-advanced-javascript/oo-challenges.md)
+- APIs via AJAX and JSON: 2 days (carries into Week 3)
   - What is AJAX?
   - jQuery's AJAX methods (ajax, get, post) & under-the-hood JavaScript's pure AJAX methods (only for educational reasons)
   - Handling success and failures with callbacks
   - Firebase (or some other competitor) configuration and consumption
-  - Promises: definition, usage, etc
```

Comment:
Correct -- but I still maintain that the way to best illustrate the utility of promises to the students is to show them how they can be implemented with respect to API calls. If we do introduce promises in the ECMA6 section, I strongly suggest we spend some time doing examples with API heavy tasks. 

Perhaps the Trello API learning outcome requires implementing all API calls with promises.

Also, for what it's worth, the [Firebase SDK](https://firebase.googleblog.com/2016/01/keeping-our-promises-and-callbacks_76.html) is A+ compatible.